pynicotine/gtkgui/widgets/textentry.py

```diff
@@ -629,13 +629,25 @@ def on_show_search_accelerator(self, *_args):
 
     def set_visible(self, visible):
 
+        self.search_bar.set_search_mode(False)
+
         if visible:
+            text_buffer = self.textview.get_buffer()
+            text_buffer_selection_bounds = text_buffer.get_selection_bounds()
+
+            if not self.entry.get_text() and text_buffer_selection_bounds:
+                selection_start, selection_end = text_buffer_selection_bounds
+
+                selection_content = text_buffer.get_text(
+                    selection_start, selection_end, include_hidden_chars=True).strip(" \n")
+
+                if "\n" not in selection_content:
+                    self.entry.set_text(selection_content)
+
             self.search_bar.set_search_mode(True)
             self.entry.grab_focus()
             return
```

Comment:
Having used some GTK-based text editors a lot, these are a couple things I expect:
- Selected text is inserted into the search entry, no matter how many lines
- Selecting a different piece of text in the text view and pressing Ctrl+F again starts a new search
- Cursor is placed at the end of search entry (ideally the text would be selected to allow typing to overwrite it, but that's not possible with the current implementation)

```suggestion
        self.search_bar.set_search_mode(visible)

        if visible:
            text_buffer = self.textview.get_buffer()
            selection_bounds = text_buffer.get_selection_bounds()

            if selection_bounds:
                selection_start, selection_end = selection_bounds
                selection_content = text_buffer.get_text(
                    selection_start, selection_end, include_hidden_chars=True)

                if self.entry.get_text().lower() != selection_content.lower():
                    self.entry.set_text(selection_content)

            self.entry.grab_focus()
            self.entry.set_position(-1)
            return
```